File: frontend/Gorki/src/app/service/rating-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../env/environment';
import { Observable } from 'rxjs';
import { CreateRatingDto } from '../model/ui/create-rating-dto';
import { CreatedRatingDto } from '../model/ui/created-rating-dto';
@Injectable({
  providedIn: 'root',
})
export class RatingService {
   constructor(private http: HttpClient) {}

  rateRide(rideId: number, dto: CreateRatingDto): Observable<CreatedRatingDto> {
    return this.http.post<CreatedRatingDto>(
      `${environment.apiHost}/rides/${rideId}/rating`,
      dto
    );
  }
}

File: mobile/app/src/main/java/ftn/mrs_team11_gorki/service/RatingService.java
package ftn.mrs_team11_gorki.service;

import ftn.mrs_team11_gorki.dto.CreateRatingDTO;
import ftn.mrs_team11_gorki.dto.CreatedRatingDTO;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface RatingService {

    @GET("api/rides/ratings/pending-latest")
    Call<Long> getPendingLatestRideId();

    @POST("api/rides/{id}/rating")
    Call<CreatedRatingDTO> rateRide(@Path("id") long rideId, @Body CreateRatingDTO dto);
}
